### PR TITLE
Support enabling service discovery for deployments

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details.

### CODE REVIEW

**Comment by dfarrell07** [comment]
I added tests for this on my fork but they don't currently provide any useful feedback because the consuming project E2E always uses the Shipyard GHAs from the main branch, not the (potentially updated) local ones.

https://github.com/dfarrell07/shipyard/pull/6

**Comment by dfarrell07** [comment]
This is meant to fix the errors for Lighthouse E2E in submariner-operator caused by moving to Shipyard's E2E GHA, which didn't have a lighthouse param.

```
Unexpected input(s) 'lighthouse', valid inputs are ['cabledriver', 'deploytool', 'globalnet', 'working-directory']
```

https://github.com/submariner-io/submariner-operator/actions/runs/549914869

**Comment by dfarrell07** [comment]
Bump, we need to get this merged to actually test Lighthouse in submariner-operator.

**Review by tpantelis — APPROVED** [review]

**Review by nyechiel — APPROVED** [review]

File: Makefile.inc [hunk 1]
```diff
@@ -15,6 +15,10 @@ else ifneq (,$(filter wireguard,$(_using)))
 override DEPLOY_ARGS += --cable_driver wireguard
 endif
 
+ifneq (,$(filter lighthouse,$(_using)))
+override DEPLOY_ARGS += --deploytool_broker_args '--service-discovery'
+endif
```
Comment (skitt) [inline Makefile.inc]: Service discovery is enabled by default, don’t we need

```suggestion
ifneq (,$(filter lighthouse,$(_using)))
override DEPLOY_ARGS += --deploytool_broker_args '--service-discovery'
else
override DEPLOY_ARGS += --deploytool_broker_args '--service-discovery=false'
endif
```

instead?

**Review by skitt — DISMISSED** [review]

**Comment by dfarrell07** [comment]
The Helm job failures are not related to this PR.

**Comment by dfarrell07** [comment]
@skitt Will you please re-review? I adopted your suggestion and added you as a co-author.

**Review by SteveMattar — CHANGES_REQUESTED** [review]
The flags are not supported by `helm` charts deployment.
For example: instead of `--service-discovery=false` it should be `--set broker.serviceDiscovery=false`

my suggestion:
**deploy.sh**


**deploy-helm.sh**



**deploy-operator.sh**


Then you can drop the `--deploytool_broker_args` and use this instead:

```sh
ifneq (,$(filter lighthouse,$(_using)))
override DEPLOY_ARGS +=  --serviceDiscovery
endif
```

**Comment by mangelajo** [comment]
@dfarrell07 see @SteveMattar comments above  ^

**Review by SteveMattar — APPROVED** [review]

**Pull request merged**